<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>levi - Landscape Expression Visualization Interface • levi</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css"><script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet"><script src="pkgdown.js"></script><meta property="og:title" content="levi - Landscape Expression Visualization Interface"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-title-body">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">levi</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.20.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="articles/levi.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/jrybarczyk/levi/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>levi - Landscape Expression Visualization Interface</h1>
    </div>


<p><img src="https://bioconductor.org/shields/availability/release/levi.svg"><img src="https://bioconductor.org/shields/downloads/release/levi.svg"><img src="https://bioconductor.org/shields/years-in-bioc/levi.svg"><img src="https://bioconductor.org/shields/build/release/bioc/levi.svg"><img src="https://bioconductor.org/shields/dependencies/release/levi.svg"></p>
<div id="levi---landscape-expression-visualization-interface" class="section level1">

<p><strong>Authors:</strong> José R. Pilan, Isabelle M. da Silva, Agnes A. S. Takeda, Jose L. Rybarczyk-Filho</p>
<p><strong>Maintainer:</strong> José L. Rybarczyk-Filho</p>
<div class="section level2">
<h2 id="summary">Summary<a class="anchor" aria-label="anchor" href="#summary"></a></h2>
<p>The integration of biological data emerges as a powerful tool for extracting information capable of describing the biological system in greater detail. The use of transcriptomic data has become ubiquitous among researchers in addressing various biological questions. In the past decade, we have witnessed the integration of biological network analysis to enrich and complement these responses. This study proposes the synergistic integration of transcriptomic data and networks, adopting an approach analogous to the Heatmap technique. The Landscape Expression Visualization Interface (LEVI) is an open-source package developed for the R environment, aiming to provide an enhanced visualization of gene expression projection onto a biological network.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a></h2>
<p><code>levi</code> is accessible on the bioconductor.org platform.</p>
<p>To install this package, start R and enter:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"levi"</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="alternative-installation-method-using-devtools">Alternative installation method using devtools<a class="anchor" aria-label="anchor" href="#alternative-installation-method-using-devtools"></a></h3>
<p>If you prefer, you can also install <code>levi</code> using <code>devtools</code>. This method might be useful if you want to install the development version directly from a GitHub repository or another source. To install using <code>devtools</code>, you’ll first need to ensure that <code>devtools</code> is installed. If it’s not, you can install it using the following command:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://devtools.r-lib.org/" class="external-link">"devtools"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"jrybarczyk/levi"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a></h2>
<p><strong><code>levi</code></strong> (<strong>Landscape Expression Visualization Interface</strong>) is an R package developed to enable the visualization of gene expression projections on a biological network. It leverages two main modes of interaction: a Graphical User Interface (GUI) powered by the Shiny package for an accessible, user-friendly experience, and script-based usage for advanced analysis and automation.</p>
<p>The GUI mode is designed to make the package approachable for users who may not be familiar with coding in R or prefer a visual approach to data analysis. It allows users to upload data, adjust visualization parameters, and interact with the results intuitively. This accessibility facilitates the exploration and interpretation of gene expression data within biological networks.</p>
<p>For users experienced in R or those requiring quicker, possibly automated analyses, <code>levi</code> can be operated through scripts. This method offers greater flexibility and customization, making it suitable for integrating <code>levi</code> into broader data analysis workflows. Scripting can significantly save time, especially when dealing with large datasets or conducting batch analyses.</p>
<p><code>levi</code> requires two files for use: - A file containing the expression levels of the genes (microarray , RNA-seq, Single Cell data). <a href="https://github.com/jrybarczyk/levi/blob/master/inst/extdata/expression.dat" class="external-link">See example</a>. - A file containing the biological network. <a href="https://github.com/jrybarczyk/levi/blob/master/inst/extdata/medusa.dat" class="external-link">See example</a>.</p>
</div>
<div class="section level2">
<h2 id="files">Files<a class="anchor" aria-label="anchor" href="#files"></a></h2>
<div class="section level3">
<h3 id="gene-expression-levels">Gene Expression Levels<a class="anchor" aria-label="anchor" href="#gene-expression-levels"></a></h3>
<p>This file should contain the genes of interest, previously normalized by the user. The expression file must have a column with gene identification (Gene Symbol, Entrez, etc.) and at least one column with gene expression levels (treatment, case, control, etc.) <a href="https://github.com/jrybarczyk/levi/blob/devel/inst/extdata/expression.dat" class="external-link">see example</a>. The user can compare expression levels between samples if there are more columns containing these data.</p>
<p>If the expression file does not have values for all genes in the network, a message will be displayed showing a log file path to a temporary directory with gene names. In the <strong>landscape</strong> construction, genes with no expression value will be displayed with values close to 0.5, demonstrating that there were no changes (down-regulated or up-regulated genes).</p>
<p>Datasets of gene expression can be obtained from online databases: - <a href="https://www.ncbi.nlm.nih.gov/geo/" class="external-link">Gene Expression Omnibus (GEO)</a> - <a href="https://www.ebi.ac.uk/arrayexpress/" class="external-link">Array Express</a> - <a href="https://cancergenome.nih.gov/" class="external-link">The Cancer Genome Atlas (TCGA)</a> - <a href="https://www.ncbi.nlm.nih.gov/sra" class="external-link">Sequence Read Archive (SRA)</a></p>
</div>
<div class="section level3">
<h3 id="biological-network">Biological Network<a class="anchor" aria-label="anchor" href="#biological-network"></a></h3>
<p>The <code>levi</code> supports several extensions of biological network files (*.net, *.dyn, *.txt, *.dat). It is recommended to obtain interaction data/biological associations from online repositories such as the The user should build the biological network using specific tools such as Cytoscape, RedeR, Medusa, etc. It is recommended to obtain interaction data/biological associations from online repositories: - <a href="https://string-db.org/" class="external-link">STRING database</a> - <a href="http://stitch.embl.de/" class="external-link">STITCH database</a> - <a href="http://starbase.sysu.edu.cn/" class="external-link">StarBase</a> - <a href="http://www.mirbase.org/" class="external-link">miRBase</a> - <a href="http://www.lncrnadb.org/" class="external-link">lncRNAdb</a> - <a href="http://www.lbbc.ibb.unesp.br/htri" class="external-link">HTRIdb</a></p>
<table class="table"><colgroup><col width="40%"><col width="40%"><col width="18%"></colgroup><thead><tr class="header"><th>Extension</th>
<th>File Type</th>
<th>Link</th>
</tr></thead><tbody><tr class="odd"><td>dat</td>
<td>Medusa (DAT)</td>
<td><a href="https://github.com/jrybarczyk/levi/blob/devel/inst/extdata/medusa.dat" class="external-link">Example DAT</a></td>
</tr><tr class="even"><td>dyn</td>
<td>RedeR (DYN)</td>
<td><a href="https://github.com/jrybarczyk/levi/blob/devel/inst/extdata/RedeR.dyn" class="external-link">Example DYN</a></td>
</tr><tr class="odd"><td>net</td>
<td>Pajek (NET)</td>
<td><a href="https://github.com/jrybarczyk/levi/blob/devel/inst/extdata/Pajek.net" class="external-link">Example NET</a></td>
</tr><tr class="even"><td>stg</td>
<td>STRING / STITCH</td>
<td><a href="https://github.com/jrybarczyk/levi/blob/devel/inst/extdata/string_network_coordinates.txt" class="external-link">Example coordinates</a></td>
</tr><tr class="odd"><td>stg</td>
<td>STRING / STITCH</td>
<td><a href="https://github.com/jrybarczyk/levi/blob/devel/inst/extdata/string_interactions.tsv" class="external-link">Example interactions</a></td>
</tr></tbody></table></div>
</div>
<div class="section level2">
<h2 id="viewing-modes">Viewing Modes<a class="anchor" aria-label="anchor" href="#viewing-modes"></a></h2>
<div class="section level3">
<h3 id="graphical-user-interface-gui">Graphical User Interface (GUI)<a class="anchor" aria-label="anchor" href="#graphical-user-interface-gui"></a></h3>
<p>levi is developed with a Graphical User Interface (GUI) leveraging the Shiny package, enabling it to be launched both within the RStudio environment and the user’s default web browser. Choosing Shiny for the GUI aims to provide a highly interactive experience, where users can intuitively manipulate data and visual settings. This makes the analysis of gene expression in biological networks accessible even to those who may not be familiar with programming in R, facilitating data exploration and interpretation in a visual manner.</p>
<p>Furthermore, the core analysis of levi is implemented in Rcpp (C++), a strategic decision to take advantage of the efficiency and performance this language offers. This hybrid approach ensures that while the Shiny-based GUI manages user interaction in a fluid and responsive manner, heavy data processing is optimally executed in the backend in Rcpp. This results in a powerful tool that combines ease of use with the capability to perform complex and computationally intensive analyses.</p>
<p>Every element of the GUI, from data loading to selecting parameters for analysis and visualization adjustments, has been meticulously designed to support users in conducting their analyses without needing deep programming knowledge. Simultaneously, the integration of the C++ analysis core through appropriate interfaces with the R environment allows levi to carry out advanced calculations effectively, maintaining the interactivity and accessibility provided by the Shiny interface.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/levi" class="external-link">levi</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/LEVIui.html">LEVIui</a></span><span class="op">(</span>browser<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>  <span class="co"># Launch Levi to Browser.</span></span>
<span><span class="fu"><a href="reference/LEVIui.html">LEVIui</a></span><span class="op">(</span>browser<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span> <span class="co"># Launch Levi to R environment.</span></span></code></pre></div>
<center>
<img src="https://github.com/jrybarczyk/levi/blob/master/vignettes/levi.png" width="150%"></center>
<center>
<b>Figure 2: </b> Graphical User Interface (GUI).
</center>
<ul><li>
<strong>1</strong> - <strong>Tabs</strong>
<ul><li>
<strong>File</strong> - General options for uploading the data files.</li>
<li>
<strong>Settings</strong> - Options to build the <strong>landscape</strong>.</li>
</ul></li>
<li>
<strong>2</strong> - <strong>Network input type:</strong> - Selection of the biological network input format. The options are available on the table 1.</li>
<li>
<strong>3</strong> - <strong>Upload the network file</strong> - Input button to select the biological network file.</li>
<li>
<strong>4</strong> - <strong>Upload the expression file</strong> - Input button to select the gene expression levels file.</li>
<li>
<strong>5</strong> - <strong>Expression values in log scale</strong> - select when there is low variation between expression data</li>
<li>
<strong>6</strong> - <strong>Selected fields</strong> - Selecting the type of analysis. Comparison between two samples or a single sample.</li>
<li>
<strong>7</strong> - <strong>Select gene symbol field</strong> - Selection of the gene ID contained in the gene expression levels file. The gene ID in the biological network file must be the same in the gene expression levels file. <strong>Select test field</strong>
</li>
<li>Select the case/test sample. <strong>Select control field</strong> - Select the control sample. If <strong>levi</strong> detects the case/test sample and control sample as equal, then <strong>levi</strong> will apply the “single sample” analysis.</li>
<li>
<strong>8</strong> - <strong>Chart Colors</strong> - Color palette to build the <strong>landscape</strong>. There is two options the <strong>Multicolor</strong> has 20 color levels combined. The <strong>Two colors</strong> has two categories of color palettes, multicolor and two colors. For the selection of two colors the available options are: <em>purple_pink</em>, <em>green_blue</em>, <em>blue_yellow</em>, <em>pink_green</em>, <em>orange_purple</em>, <em>green_marine</em>. Both of them is targeted to show the biological network with higher expression. (Figure 3 and Figure 4).</li>
<li>
<strong>9</strong> - <strong>Chart with contour</strong> - Enable or disable the contour lines in the <strong>landscape</strong>.</li>
<li>
<strong>10</strong> - <strong>landscape</strong> Image display area. The user can select specific areas of the image to inspect the gene name and position.</li>
<li>
<strong>11</strong> - <strong>Expression area:</strong> - Total expression value of the selected area in the image.</li>
<li>
<strong>12</strong> - <strong>Download options</strong>
<ul><li>
<strong>File format to landscape</strong> - Select the output format of the <strong>landscape</strong>. The options available are: TIFF, BMP, JPEG and PNG. We recommend using this option in the browser.</li>
<li>
<strong>Download Plot</strong> - Button to save the file.</li>
</ul></li>
<li>
<strong>13</strong> - Visualization of gene name and expression value of the selected area in the <strong>landscape</strong>. The user can save the table in a csv file (button <strong>Download Data</strong>)</li>
<li>
<strong>14</strong> - <strong>Settings</strong>
<ul><li>
<strong>Contrast</strong> - Constrast value in the <strong>landscape</strong>. The variable range is 1 to 100. The default value is 50.</li>
<li>
<strong>Resolution</strong> - Image size of the <strong>landscape</strong>. The variable range is 1 to 100. The default value is 50. If this parameter is higher, then the total time required will be longer.</li>
<li>
<strong>Smoothing</strong> - Smoothing of the <strong>landscape</strong>. The variable range is 1 to 100. The default value is 50.I f this parameter is higher, then the total time required will be longer.</li>
<li>
<strong>Zoom</strong> - Zoom value for the <strong>landscape</strong>. The variable range is 1 to 100. The default value is 50.</li>
</ul></li>
</ul></div>
<div class="section level3">
<h3 id="script">Script<a class="anchor" aria-label="anchor" href="#script"></a></h3>
<p>The script mode offers enhanced flexibility for more detailed settings adjustments and comprehensive automation. This mode is particularly well-suited for advanced users who require precise control over their analysis parameters and workflows. Additionally, it is ideal for processing large datasets in bulk, enabling efficient data handling and analysis customization. This capability makes it an invaluable tool for researchers and data scientists looking to streamline their data analysis pipelines, ensuring both scalability and reproducibility in their work.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/levi" class="external-link">levi</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">template_network</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"levi"</span><span class="op">)</span>,<span class="st">"extdata"</span>,</span>
<span>                                <span class="st">"medusa.dat"</span>, fsep <span class="op">=</span> <span class="va">.Platform</span><span class="op">$</span><span class="va">file.sep</span><span class="op">)</span></span>
<span></span>
<span><span class="va">template_expression</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"levi"</span><span class="op">)</span>,</span>
<span>                                <span class="st">"extdata"</span>,<span class="st">"expression.dat"</span>, </span>
<span>                                fsep <span class="op">=</span> <span class="va">.Platform</span><span class="op">$</span><span class="va">file.sep</span><span class="op">)</span></span>
<span></span>
<span><span class="va">multicolor</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/levi.html">levi</a></span><span class="op">(</span>networkCoordinatesInput <span class="op">=</span> <span class="va">template_network</span>,</span>
<span>                expressionInput <span class="op">=</span> <span class="va">template_expression</span>, fileTypeInput <span class="op">=</span> <span class="st">"dat"</span>,</span>
<span>                geneSymbolnput <span class="op">=</span> <span class="st">"ID"</span>, </span>
<span>                readExpColumn<span class="op">=</span></span>
<span>                <span class="fu"><a href="reference/readExpColumn.html">readExpColumn</a></span><span class="op">(</span><span class="st">"TumorCurrentSmoker-NormalNeverSmoker"</span><span class="op">)</span>, </span>
<span>                contrastValueInput <span class="op">=</span> <span class="fl">50</span>, resolutionValueInput  <span class="op">=</span> <span class="fl">50</span>, </span>
<span>                zoomValueInput <span class="op">=</span> <span class="fl">50</span>, smoothValueInput <span class="op">=</span> <span class="fl">50</span>, contourLevi <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">twocolors</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/levi.html">levi</a></span><span class="op">(</span>networkCoordinatesInput <span class="op">=</span> <span class="va">template_network</span>,</span>
<span>                expressionInput <span class="op">=</span> <span class="va">template_expression</span>, fileTypeInput <span class="op">=</span> <span class="st">"dat"</span>,</span>
<span>                geneSymbolnput <span class="op">=</span> <span class="st">"ID"</span>, </span>
<span>                readExpColumn<span class="op">=</span></span>
<span>                <span class="fu"><a href="reference/readExpColumn.html">readExpColumn</a></span><span class="op">(</span><span class="st">"TumorCurrentSmoker-NormalNeverSmoker"</span><span class="op">)</span>,</span>
<span>                setcolor <span class="op">=</span> <span class="st">"pink_green"</span>, contourLevi <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>The script mode allows the user to compare combinations between two experiments in the gene expression levels file. The readExpColumn function can be used to this task to inform the combination separating by dash (-) and to add more combinations separate by comma (,).</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/levi" class="external-link">levi</a></span><span class="op">)</span></span>
<span><span class="va">base</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/readExpColumn.html">readExpColumn</a></span><span class="op">(</span><span class="st">"TumorFormerSmoker-NormalFormerSmoker"</span>, </span>
<span>                        <span class="st">"TumorNeverSmoker-TumorNeverSmoker"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">template_network</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"levi"</span><span class="op">)</span>,<span class="st">"extdata"</span>,</span>
<span>                                <span class="st">"medusa.dat"</span>, fsep <span class="op">=</span> <span class="va">.Platform</span><span class="op">$</span><span class="va">file.sep</span><span class="op">)</span></span>
<span></span>
<span><span class="va">template_expression</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"levi"</span><span class="op">)</span>,</span>
<span>                                <span class="st">"extdata"</span>,<span class="st">"expression.dat"</span>, </span>
<span>                                fsep <span class="op">=</span> <span class="va">.Platform</span><span class="op">$</span><span class="va">file.sep</span><span class="op">)</span></span>
<span></span>
<span><span class="va">multicolor</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/levi.html">levi</a></span><span class="op">(</span>networkCoordinatesInput <span class="op">=</span> <span class="va">template_network</span>,</span>
<span>                    expressionInput <span class="op">=</span> <span class="va">template_expression</span>, </span>
<span>                    fileTypeInput <span class="op">=</span> <span class="st">"dat"</span>,</span>
<span>                    geneSymbolnput <span class="op">=</span> <span class="st">"ID"</span>, readExpColumn<span class="op">=</span> <span class="va">base</span>, </span>
<span>                    contrastValueInput <span class="op">=</span> <span class="fl">50</span>, resolutionValueInput  <span class="op">=</span> <span class="fl">50</span>, </span>
<span>                    zoomValueInput <span class="op">=</span> <span class="fl">50</span>, smoothValueInput <span class="op">=</span> <span class="fl">50</span>, </span>
<span>                    contourLevi <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">twocolors</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/levi.html">levi</a></span><span class="op">(</span>networkCoordinatesInput <span class="op">=</span> <span class="va">template_network</span>,</span>
<span>                expressionInput <span class="op">=</span> <span class="va">template_expression</span>, fileTypeInput <span class="op">=</span> <span class="st">"dat"</span>,</span>
<span>                geneSymbolnput <span class="op">=</span> <span class="st">"ID"</span>, </span>
<span>                readExpColumn<span class="op">=</span> <span class="va">base</span>,</span>
<span>                setcolor <span class="op">=</span> <span class="st">"pink_green"</span>, contourLevi <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><code>levi</code> offers a versatile solution for gene expression and biological network analysis, catering to a wide range of user preferences and technical expertise levels. <a href="https://github.com/jrybarczyk/levi/blob/paper/Case/Case.pdf" class="external-link">more examples</a></p>
</div>
</div>
<div class="section level2">
<h2 id="information-about-downloads-of-levi-on-bioconductororg">Information about downloads of levi on Bioconductor.org<a class="anchor" aria-label="anchor" href="#information-about-downloads-of-levi-on-bioconductororg"></a></h2>
<p><img src="https://github.com/jrybarczyk/levi/blob/paper/miscellaneous/Rplot.png"></p>
</div>
</div>


  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>

</div>



      <footer><div class="copyright">
  <p></p><p>Developed by Rafael Pilan, Jose Rybarczyk Filho.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

      </footer></div>






  </body></html>

