<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>levi — levi • levi</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="levi — levi"><meta property="og:description" content="This is the Levi script mode. It allows you to create the
integration of networks and gene expression levels as batch
processing"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">levi</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.20.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/levi.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/jrybarczyk/levi/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>levi</h1>
    <small class="dont-index">Source: <a href="https://github.com/jrybarczyk/levi/blob/master/R/levi.R" class="external-link"><code>R/levi.R</code></a></small>
    <div class="hidden name"><code>levi.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This is the Levi script mode. It allows you to create the
integration of networks and gene expression levels as batch
processing</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">levi</span><span class="op">(</span><span class="va">expressionInput</span>, <span class="va">fileTypeInput</span>, <span class="va">networkCoordinatesInput</span>,</span>
<span><span class="va">networkInteractionsInput</span>, <span class="va">geneSymbolnput</span>, <span class="va">readExpColumn</span>,</span>
<span><span class="va">contrastValueInput</span>, <span class="va">zoomValueInput</span>, <span class="va">resolutionValueInput</span>,</span>
<span><span class="va">smoothValueInput</span>, <span class="va">expressionLog</span>, <span class="va">contourLevi</span>, <span class="va">setcolor</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-expressioninput">expressionInput<a class="anchor" aria-label="anchor" href="#arg-expressioninput"></a></dt>
<dd><p>Filename of gene expression data, which is a
numeric data.frame or matrix. The rows represent genes/proteins and the
columns represent the experiment (RNA-seq, microarray, etc).</p></dd>


<dt id="arg-filetypeinput">fileTypeInput<a class="anchor" aria-label="anchor" href="#arg-filetypeinput"></a></dt>
<dd><p>Filename of biological network. Levi can read files
written in the following formats: Medusa (DAT), RedeR (DYN), Pajek (NET)
and STRING/STITCH</p></dd>


<dt id="arg-networkcoordinatesinput">networkCoordinatesInput<a class="anchor" aria-label="anchor" href="#arg-networkcoordinatesinput"></a></dt>
<dd><p>It allows the user to load the coordinate
of the nodes the network.</p></dd>


<dt id="arg-networkinteractionsinput">networkInteractionsInput<a class="anchor" aria-label="anchor" href="#arg-networkinteractionsinput"></a></dt>
<dd><p>Parameter available only to
STRING/STITCH data format.
It allows the user to load the interaction data file of the network.</p></dd>


<dt id="arg-genesymbolnput">geneSymbolnput<a class="anchor" aria-label="anchor" href="#arg-genesymbolnput"></a></dt>
<dd><p>Column name from gene expression data containing the
identifier (gene Symbol, Entrez ID, EMSEMBL, etc).</p></dd>


<dt id="arg-readexpcolumn">readExpColumn<a class="anchor" aria-label="anchor" href="#arg-readexpcolumn"></a></dt>
<dd><p>Variable from readExpColumn function containing the
comparisons of the experiments</p></dd>


<dt id="arg-contrastvalueinput">contrastValueInput<a class="anchor" aria-label="anchor" href="#arg-contrastvalueinput"></a></dt>
<dd><p>Numeric value for image contrast. The variable
range is 0 to 100. The default value is 50</p></dd>


<dt id="arg-zoomvalueinput">zoomValueInput<a class="anchor" aria-label="anchor" href="#arg-zoomvalueinput"></a></dt>
<dd><p>Numeric value for image zoom. The variable range is 0
to 100. The default value is 50.</p></dd>


<dt id="arg-resolutionvalueinput">resolutionValueInput<a class="anchor" aria-label="anchor" href="#arg-resolutionvalueinput"></a></dt>
<dd><p>Numeric value for image resolution. The variable
range is 0 to 100. The default value is 50.</p></dd>


<dt id="arg-smoothvalueinput">smoothValueInput<a class="anchor" aria-label="anchor" href="#arg-smoothvalueinput"></a></dt>
<dd><p>Numeric value for image smoothness. The variable
range is 0 to 100. The default is 50.</p></dd>


<dt id="arg-expressionlog">expressionLog<a class="anchor" aria-label="anchor" href="#arg-expressionlog"></a></dt>
<dd><p>Logical variable to indicate Log2 normalization in the
expression levels. The default is FALSE</p></dd>


<dt id="arg-contourlevi">contourLevi<a class="anchor" aria-label="anchor" href="#arg-contourlevi"></a></dt>
<dd><p>Logical variable to allow contour lines. The default is
TRUE.</p></dd>


<dt id="arg-setcolor">setcolor<a class="anchor" aria-label="anchor" href="#arg-setcolor"></a></dt>
<dd><p>Select the color palette to build the heatmat. There is
two options the **Multicolor** has 20 color levels combined. The
**Two colors** has two types of color and the options available are: *purple_pink*,
*green_blue*, *blue_yellow*, *pink_green*, *orange_purple*, *green_marine*.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>Return a ggplot object and print a image (heatmat).</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Integrates the biological network and gene expression levels
(or other type of data)</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Isabelle Mira da Silva (isabelle.silva@unesp.br), José Rafael Pilan (rafael.pilan@unesp.br)</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">template_network</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"levi"</span><span class="op">)</span>,<span class="st">"extdata"</span>,</span></span>
<span class="r-in"><span>   <span class="st">"medusa.dat"</span>, fsep <span class="op">=</span> <span class="va">.Platform</span><span class="op">$</span><span class="va">file.sep</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">template_expression</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"levi"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>   <span class="st">"extdata"</span>,<span class="st">"expression.dat"</span>, fsep <span class="op">=</span> <span class="va">.Platform</span><span class="op">$</span><span class="va">file.sep</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">multicolor</span> <span class="op">&lt;-</span> <span class="fu">levi</span><span class="op">(</span>networkCoordinatesInput <span class="op">=</span> <span class="va">template_network</span>,</span></span>
<span class="r-in"><span>   expressionInput <span class="op">=</span> <span class="va">template_expression</span>, fileTypeInput <span class="op">=</span> <span class="st">"dat"</span>,</span></span>
<span class="r-in"><span>   geneSymbolnput <span class="op">=</span> <span class="st">"ID"</span>,</span></span>
<span class="r-in"><span>   readExpColumn<span class="op">=</span><span class="fu"><a href="readExpColumn.html">readExpColumn</a></span><span class="op">(</span><span class="st">"TumorCurrentSmoker-NormalNeverSmoker"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>   contrastValueInput <span class="op">=</span> <span class="fl">50</span>, resolutionValueInput  <span class="op">=</span> <span class="fl">50</span>, zoomValueInput <span class="op">=</span> <span class="fl">50</span>,</span></span>
<span class="r-in"><span>   smoothValueInput <span class="op">=</span> <span class="fl">50</span>, expressionLog <span class="op">=</span> <span class="cn">FALSE</span>, contourLevi <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Coordinate system already present. Adding new coordinate system, which will</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> replace the existing one.</span>
<span class="r-plt img"><img src="levi-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">twocolors</span> <span class="op">&lt;-</span> <span class="fu">levi</span><span class="op">(</span>networkCoordinatesInput <span class="op">=</span> <span class="va">template_network</span>,</span></span>
<span class="r-in"><span>   expressionInput <span class="op">=</span> <span class="va">template_expression</span>, fileTypeInput <span class="op">=</span> <span class="st">"dat"</span>,</span></span>
<span class="r-in"><span>   geneSymbolnput <span class="op">=</span> <span class="st">"ID"</span>,</span></span>
<span class="r-in"><span>   readExpColumn <span class="op">=</span> <span class="fu"><a href="readExpColumn.html">readExpColumn</a></span><span class="op">(</span><span class="st">"TumorCurrentSmoker-NormalNeverSmoker"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>   setcolor <span class="op">=</span> <span class="st">"pink_green"</span>, contourLevi <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Coordinate system already present. Adding new coordinate system, which will</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> replace the existing one.</span>
<span class="r-plt img"><img src="levi-2.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Rafael Pilan, Jose Rybarczyk Filho.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer></div>






  </body></html>

