<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>levi - Landscape Expression Visualization Interface • levi</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="levi - Landscape Expression Visualization Interface">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">levi</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.20.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/levi.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jrybarczyk/levi/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>levi - Landscape Expression Visualization
Interface</h1>
                        <h4 data-toc-skip class="author">Isabelle M. da
Silva, José R. Pilan, Agnes A. S. Takeda, Jose L. Rybarczyk-Filho</h4>
            
            <h4 data-toc-skip class="date">27 August 2025</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/jrybarczyk/levi/blob/master/vignettes/levi.Rmd" class="external-link"><code>vignettes/levi.Rmd</code></a></small>
      <div class="hidden name"><code>levi.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>The <strong>levi (Landscape Expression Visualization
Interface)</strong> is a package for the R environment, developed to
enable the visualization of the projection of gene expression on a
biological network. <strong>levi</strong> is based on two other
software, the first is the Viacomplex (<span class="citation">Castro et
al. (2009)</span>), which was written in Fortran using the Dislin
library to plot the data. The second is Galant (<span class="citation">Camilo et al. (2013)</span>), which corresponds to a
plugin for Cytoscape software.</p>
<p>Two files are required to use <strong>levi</strong>: The file
containing the expression levels of the genes and a file containing the
biological network.</p>
</div>
<div class="section level2">
<h2 id="files">Files<a class="anchor" aria-label="anchor" href="#files"></a>
</h2>
<div class="section level3">
<h3 id="gene-expression-levels">Gene Expression Levels<a class="anchor" aria-label="anchor" href="#gene-expression-levels"></a>
</h3>
<p>This file should contain the genes of interest previously normalized
by the user. The expression file (see <a href="https://bit.ly/2KSBaOo" class="external-link">Example</a>) must have a column with the
identification of genes (Gene Symbol, Entrez, etc.) and at least one
column with gene expression levels (treatment, case, control, etc.). The
user can compare the expression levels between samples if there are more
columns containing these data.</p>
<p>If the expression file does not have values for all the genes of the
network, a message will be displayed showing log file path to a
temporary directory with the gene names. In the
<strong>landscape</strong> construction, genes with no expression value
will be displayed with values close to 0.5 demonstrating that there were
no changes (down-regulated or up-regulated genes).</p>
<p>Data sets of gene expression can be obtained from online
databases:</p>
<ul>
<li><a href="https://www.ncbi.nlm.nih.gov/geo/" class="external-link">Gene Expression Omnibus
(GEO)</a></li>
<li><a href="https://www.ebi.ac.uk/arrayexpress/" class="external-link">Array Express</a></li>
<li><a href="https://cancergenome.nih.gov/" class="external-link">The Cancer Genome Atlas
(TCGA)</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/sra" class="external-link">Sequence Read Archive
(SRA)</a></li>
</ul>
</div>
<div class="section level3">
<h3 id="biological-network">Biological Network<a class="anchor" aria-label="anchor" href="#biological-network"></a>
</h3>
<p>The <strong>levi</strong> supports several extensions of biological
network files (*.net, *.dyn, *.txt, *.dat). The user should build the
biological network using specific tools such as: Cytoscape, RedeR,
Medusa, etc. To build the biological network is recommended the
obtention of interaction data/ biological associations from online
repositories:</p>
<ul>
<li><a href="https://string-db.org/" class="external-link">STRING database</a></li>
<li><a href="http://starbase.sysu.edu.cn/" class="external-link">StarBase</a></li>
<li><a href="http://www.mirbase.org/" class="external-link">miRBase</a></li>
<li><a href="http://www.lncrnadb.org/" class="external-link">lncRNAdb</a></li>
<li><a href="http://www.lbbc.ibb.unesp.br/htri" class="external-link">HTRIdb</a></li>
</ul>
<table class="table">
<colgroup>
<col width="13%">
<col width="22%">
<col width="63%">
</colgroup>
<thead><tr class="header">
<th align="center">Extension</th>
<th align="center">File Type</th>
<th align="center">Link</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="center">dat</td>
<td align="center">Medusa (DAT)</td>
<td align="center"><a href="https://bit.ly/2sj6RJh" class="external-link">Example DAT</a></td>
</tr>
<tr class="even">
<td align="center">dyn</td>
<td align="center">RedeR (DYN)</td>
<td align="center"><a href="https://bit.ly/2IJGaZ7" class="external-link">Example DYN</a></td>
</tr>
<tr class="odd">
<td align="center">net</td>
<td align="center">Pajek (NET)</td>
<td align="center"><a href="https://bit.ly/2IKzaLv" class="external-link">Example NET</a></td>
</tr>
<tr class="even">
<td align="center">stg</td>
<td align="center">STRING / STITCH</td>
<td align="center"><a href="https://bit.ly/2xjnaLU" class="external-link">Example
coordinates</a></td>
</tr>
<tr class="odd">
<td align="center">stg</td>
<td align="center">STRING / STITCH</td>
<td align="center"><a href="https://bit.ly/2J2OToH" class="external-link">Example
interactions</a></td>
</tr>
</tbody>
</table>
<p>#Viewing Modes</p>
<p><strong>levi</strong> has two viewing modes: <strong>Graphical User
Interface (GUI)</strong> and <strong>script</strong>.</p>
</div>
<div class="section level3">
<h3 id="graphical-user-interface-gui">Graphical User Interface (GUI)<a class="anchor" aria-label="anchor" href="#graphical-user-interface-gui"></a>
</h3>
<p>The GUI mode was developed using Shiny package (Figure 1). This
viewing mode can be used in the R environment (browser=FALSE) or in the
user’s default operating system browser (browser=TRUE).</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/levi" class="external-link">levi</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/LEVIui.html">LEVIui</a></span><span class="op">(</span>browser<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>  <span class="co">#Launch Levi to Browser.</span></span>
<span><span class="fu"><a href="../reference/LEVIui.html">LEVIui</a></span><span class="op">(</span>browser<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span> <span class="co">#Launch Levi to R environment.</span></span></code></pre></div>
<center>
<img src="Newlevi.png" width="150%">
</center>
<center>
<b>Figure 2: </b> Graphical User Interface (GUI).
</center>
<ul>
<li>
<strong>1</strong> - <strong>Tabs</strong>
<ul>
<li>
<strong>File</strong> - General options for uploading the data
files.</li>
<li>
<strong>Settings</strong> - Options to build the
<strong>landscape</strong>.</li>
</ul>
</li>
<li>
<strong>2</strong> - <strong>Network input type:</strong> -
Selection of the biological network input format. The options are
available on the table 1.</li>
<li>
<strong>3</strong> - <strong>Upload the network file</strong> -
Input button to select the biological network file.</li>
<li>
<strong>4</strong> - <strong>Upload the expression file</strong> -
Input button to select the gene expression levels file.</li>
<li>
<strong>5</strong> - <strong>Expression values in log scale</strong>
- select when there is low variation between expression data</li>
<li>
<strong>6</strong> - <strong>Selected fields</strong> - Selecting
the type of analysis. Comparison between two samples or a single
sample.</li>
<li>
<strong>7</strong> - <strong>Select gene symbol field</strong> -
Selection of the gene ID contained in the gene expression levels file.
The gene ID in the biological network file must be the same in the gene
expression levels file. <strong>Select test field</strong>
</li>
<li>Select the case/test sample. <strong>Select control field</strong> -
Select the control sample. If <strong>levi</strong> detects the
case/test sample and control sample as equal, then <strong>levi</strong>
will apply the “single sample” analysis.</li>
<li>
<strong>8</strong> - <strong>Chart Colors</strong> - Color palette
to build the <strong>landscape</strong>. There is two options the
<strong>Multicolor</strong> has 20 color levels combined. The
<strong>Two colors</strong> has two categories of color palettes,
multicolor and two colors. For the selection of two colors the available
options are: <em>purple_pink</em>, <em>green_blue</em>,
<em>blue_yellow</em>, <em>pink_green</em>, <em>orange_purple</em>,
<em>green_marine</em>. Both of them is targeted to show the biological
network with higher expression. (Figure 3 and Figure 4).</li>
<li>
<strong>9</strong> - <strong>Chart with contour</strong> - Enable or
disable the contour lines in the <strong>landscape</strong>.</li>
<li>
<strong>10</strong> - <strong>landscape</strong> Image display area.
The user can select specific areas of the image to inspect the gene name
and position.</li>
<li>
<strong>11</strong> - <strong>Expression area:</strong> - Total
expression value of the selected area in the image.</li>
<li>
<strong>12</strong> - <strong>Download options</strong>
<ul>
<li>
<strong>File format to landscape</strong> - Select the output format
of the <strong>landscape</strong>. The options available are: TIFF, BMP,
JPEG and PNG. We recommend using this option in the browser.</li>
<li>
<strong>Download Plot</strong> - Button to save the file.</li>
</ul>
</li>
<li>
<strong>13</strong> - Visualization of gene name and expression
value of the selected area in the <strong>landscape</strong>. The user
can save the table in a csv file (button <strong>Download
Data</strong>)</li>
<li>
<strong>14</strong> - <strong>Settings</strong>
<ul>
<li>
<strong>Contrast</strong> - Constrast value in the
<strong>landscape</strong>. The variable range is 1 to 100. The default
value is 50.</li>
<li>
<strong>Resolution</strong> - Image size of the
<strong>landscape</strong>. The variable range is 1 to 100. The default
value is 50. If this parameter is higher, then the total time required
will be longer.</li>
<li>
<strong>Smoothing</strong> - Smoothing of the
<strong>landscape</strong>. The variable range is 1 to 100. The default
value is 50.I f this parameter is higher, then the total time required
will be longer.</li>
<li>
<strong>Zoom</strong> - Zoom value for the
<strong>landscape</strong>. The variable range is 1 to 100. The default
value is 50.</li>
</ul>
</li>
</ul>
</div>
<div class="section level3">
<h3 id="script">
<em>Script</em><a class="anchor" aria-label="anchor" href="#script"></a>
</h3>
<p>The <strong>levi</strong> scripting mode also has the settings for
<strong>landscape</strong> building (see example below).</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/levi" class="external-link">levi</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">template_network</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"levi"</span><span class="op">)</span>,<span class="st">"extdata"</span>,</span>
<span>                                <span class="st">"medusa.dat"</span>, fsep <span class="op">=</span> <span class="va">.Platform</span><span class="op">$</span><span class="va">file.sep</span><span class="op">)</span></span>
<span></span>
<span><span class="va">template_expression</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"levi"</span><span class="op">)</span>,</span>
<span>                                <span class="st">"extdata"</span>,<span class="st">"expression.dat"</span>, </span>
<span>                                fsep <span class="op">=</span> <span class="va">.Platform</span><span class="op">$</span><span class="va">file.sep</span><span class="op">)</span></span>
<span></span>
<span><span class="va">multicolor</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/levi.html">levi</a></span><span class="op">(</span>networkCoordinatesInput <span class="op">=</span> <span class="va">template_network</span>,</span>
<span>                expressionInput <span class="op">=</span> <span class="va">template_expression</span>, fileTypeInput <span class="op">=</span> <span class="st">"dat"</span>,</span>
<span>                geneSymbolnput <span class="op">=</span> <span class="st">"ID"</span>, </span>
<span>                readExpColumn<span class="op">=</span></span>
<span>                <span class="fu"><a href="../reference/readExpColumn.html">readExpColumn</a></span><span class="op">(</span><span class="st">"TumorCurrentSmoker-NormalNeverSmoker"</span><span class="op">)</span>, </span>
<span>                contrastValueInput <span class="op">=</span> <span class="fl">50</span>, resolutionValueInput  <span class="op">=</span> <span class="fl">50</span>, </span>
<span>                zoomValueInput <span class="op">=</span> <span class="fl">50</span>, smoothValueInput <span class="op">=</span> <span class="fl">50</span>, contourLevi <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Coordinate system already present. Adding new coordinate system, which will</span></span>
<span><span class="co">## replace the existing one.</span></span></code></pre>
<p><img src="levi_files/figure-html/unnamed-chunk-2-1.png" width="576"></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">twocolors</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/levi.html">levi</a></span><span class="op">(</span>networkCoordinatesInput <span class="op">=</span> <span class="va">template_network</span>,</span>
<span>                expressionInput <span class="op">=</span> <span class="va">template_expression</span>, fileTypeInput <span class="op">=</span> <span class="st">"dat"</span>,</span>
<span>                geneSymbolnput <span class="op">=</span> <span class="st">"ID"</span>, </span>
<span>                readExpColumn<span class="op">=</span></span>
<span>                <span class="fu"><a href="../reference/readExpColumn.html">readExpColumn</a></span><span class="op">(</span><span class="st">"TumorCurrentSmoker-NormalNeverSmoker"</span><span class="op">)</span>,</span>
<span>                setcolor <span class="op">=</span> <span class="st">"pink_green"</span>, contourLevi <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Coordinate system already present. Adding new coordinate system, which will</span></span>
<span><span class="co">## replace the existing one.</span></span></code></pre>
<p><img src="levi_files/figure-html/unnamed-chunk-2-2.png" width="576">
The script mode allows the user to compare combinations between two
experiments in the gene expression levels file. The readExpColumn
function can be used to this task to inform the combination separating
by dash (-) and to add more combinations separate by comma (,).</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/levi" class="external-link">levi</a></span><span class="op">)</span></span>
<span><span class="va">base</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/readExpColumn.html">readExpColumn</a></span><span class="op">(</span><span class="st">"TumorFormerSmoker-NormalFormerSmoker"</span>, </span>
<span>                        <span class="st">"TumorNeverSmoker-TumorNeverSmoker"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">template_network</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"levi"</span><span class="op">)</span>,<span class="st">"extdata"</span>,</span>
<span>                                <span class="st">"medusa.dat"</span>, fsep <span class="op">=</span> <span class="va">.Platform</span><span class="op">$</span><span class="va">file.sep</span><span class="op">)</span></span>
<span></span>
<span><span class="va">template_expression</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span>package<span class="op">=</span><span class="st">"levi"</span><span class="op">)</span>,</span>
<span>                                <span class="st">"extdata"</span>,<span class="st">"expression.dat"</span>, </span>
<span>                                fsep <span class="op">=</span> <span class="va">.Platform</span><span class="op">$</span><span class="va">file.sep</span><span class="op">)</span></span>
<span></span>
<span><span class="va">multicolor</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/levi.html">levi</a></span><span class="op">(</span>networkCoordinatesInput <span class="op">=</span> <span class="va">template_network</span>,</span>
<span>                    expressionInput <span class="op">=</span> <span class="va">template_expression</span>, </span>
<span>                    fileTypeInput <span class="op">=</span> <span class="st">"dat"</span>,</span>
<span>                    geneSymbolnput <span class="op">=</span> <span class="st">"ID"</span>, readExpColumn<span class="op">=</span> <span class="va">base</span>, </span>
<span>                    contrastValueInput <span class="op">=</span> <span class="fl">50</span>, resolutionValueInput  <span class="op">=</span> <span class="fl">50</span>, </span>
<span>                    zoomValueInput <span class="op">=</span> <span class="fl">50</span>, smoothValueInput <span class="op">=</span> <span class="fl">50</span>, </span>
<span>                    contourLevi <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Coordinate system already present. Adding new coordinate system, which will</span></span>
<span><span class="co">## replace the existing one.</span></span></code></pre>
<p><img src="levi_files/figure-html/unnamed-chunk-3-1.png" width="576"></p>
<pre><code><span><span class="co">## Coordinate system already present. Adding new coordinate system, which will</span></span>
<span><span class="co">## replace the existing one.</span></span></code></pre>
<p><img src="levi_files/figure-html/unnamed-chunk-3-2.png" width="576"></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">twocolors</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/levi.html">levi</a></span><span class="op">(</span>networkCoordinatesInput <span class="op">=</span> <span class="va">template_network</span>,</span>
<span>                expressionInput <span class="op">=</span> <span class="va">template_expression</span>, fileTypeInput <span class="op">=</span> <span class="st">"dat"</span>,</span>
<span>                geneSymbolnput <span class="op">=</span> <span class="st">"ID"</span>, </span>
<span>                readExpColumn<span class="op">=</span> <span class="va">base</span>,</span>
<span>                setcolor <span class="op">=</span> <span class="st">"pink_green"</span>, contourLevi <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Coordinate system already present. Adding new coordinate system, which will</span></span>
<span><span class="co">## replace the existing one.</span></span></code></pre>
<p><img src="levi_files/figure-html/unnamed-chunk-3-3.png" width="576"></p>
<pre><code><span><span class="co">## Coordinate system already present. Adding new coordinate system, which will</span></span>
<span><span class="co">## replace the existing one.</span></span></code></pre>
<p><img src="levi_files/figure-html/unnamed-chunk-3-4.png" width="576"></p>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<pre><code><span><span class="co">## R version 4.5.1 (2025-06-13)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 24.04.2 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] levi_1.20.0      BiocStyle_2.36.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] sass_0.4.10          generics_0.1.4       xml2_1.4.0          </span></span>
<span><span class="co">##  [4] stringi_1.8.7        digest_0.6.37        magrittr_2.0.3      </span></span>
<span><span class="co">##  [7] evaluate_1.0.4       grid_4.5.1           RColorBrewer_1.1-3  </span></span>
<span><span class="co">## [10] bookdown_0.44        fastmap_1.2.0        plyr_1.8.9          </span></span>
<span><span class="co">## [13] jsonlite_2.0.0       brio_1.1.5           promises_1.3.3      </span></span>
<span><span class="co">## [16] BiocManager_1.30.26  httr_1.4.7           scales_1.4.0        </span></span>
<span><span class="co">## [19] isoband_0.2.7        textshaping_1.0.1    jquerylib_0.1.4     </span></span>
<span><span class="co">## [22] shinydashboard_0.7.3 cli_3.6.5            shiny_1.11.1        </span></span>
<span><span class="co">## [25] rlang_1.1.6          withr_3.0.2          cachem_1.1.0        </span></span>
<span><span class="co">## [28] yaml_2.3.10          tools_4.5.1          reshape2_1.4.4      </span></span>
<span><span class="co">## [31] dplyr_1.1.4          colorspace_2.1-1     ggplot2_3.5.2       </span></span>
<span><span class="co">## [34] httpuv_1.6.16        DT_0.33              mime_0.13           </span></span>
<span><span class="co">## [37] vctrs_0.6.5          R6_2.6.1             lifecycle_1.0.4     </span></span>
<span><span class="co">## [40] stringr_1.5.1        fs_1.6.6             htmlwidgets_1.6.4   </span></span>
<span><span class="co">## [43] shinyjs_2.1.0        ragg_1.4.0           pkgconfig_2.0.3     </span></span>
<span><span class="co">## [46] desc_1.4.3           pkgdown_2.1.3        pillar_1.11.0       </span></span>
<span><span class="co">## [49] bslib_0.9.0          later_1.4.3          gtable_0.3.6        </span></span>
<span><span class="co">## [52] glue_1.8.0           Rcpp_1.1.0           systemfonts_1.2.3   </span></span>
<span><span class="co">## [55] xfun_0.53            tibble_3.3.0         tidyselect_1.2.1    </span></span>
<span><span class="co">## [58] knitr_1.50           xtable_1.8-4         farver_2.1.2        </span></span>
<span><span class="co">## [61] htmltools_0.5.8.1    igraph_2.1.4         labeling_0.4.3      </span></span>
<span><span class="co">## [64] rmarkdown_2.29       testthat_3.2.3       compiler_4.5.1</span></span></code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="reference">Reference<a class="anchor" aria-label="anchor" href="#reference"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-Camilo2013" class="csl-entry">
Camilo, Esther, Luiz A Bovolenta, Marcio L Acencio, Jose L
Rybarczyk-Filho, Mauro AA Castro, Jose CF Moreira, and Ney Lemke. 2013.
<span>“GALANT: A Cytoscape Plugin for Visualizing Data as Functional
Landscapes Projected onto Biological Networks.”</span>
<em>Bioinformatics</em> 29 (19): 2505–6.
</div>
<div id="ref-Castro2009" class="csl-entry">
Castro, Mauro AA, José L Rybarczyk Filho, Rodrigo JS Dalmolin, Marialva
Sinigaglia, José CF Moreira, José CM Mombach, and Rita MC de Almeida.
2009. <span>“ViaComplex: Software for Landscape Analysis of Gene
Expression Networks in Genomic Context.”</span> <em>Bioinformatics</em>
25 (11): 1468–69.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Rafael Pilan, Jose Rybarczyk Filho.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer>
</div>






  </body>
</html>
